tools/perspective-scripts/workspace.mjs

```diff
@@ -0,0 +1,53 @@
+// ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+// ┃ ██████ ██████ ██████       █      █      █      █      █ █▄  ▀███ █       ┃
+// ┃ ▄▄▄▄▄█ █▄▄▄▄▄ ▄▄▄▄▄█  ▀▀▀▀▀█▀▀▀▀▀ █ ▀▀▀▀▀█ ████████▌▐███ ███▄  ▀█ █ ▀▀▀▀▀ ┃
+// ┃ █▀▀▀▀▀ █▀▀▀▀▀ █▀██▀▀ ▄▄▄▄▄ █ ▄▄▄▄▄█ ▄▄▄▄▄█ ████████▌▐███ █████▄   █ ▄▄▄▄▄ ┃
+// ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+// ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+// ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+// ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+// ┃ This file is part of the Perspective library, distributed under the terms ┃
+// ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+// ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+import fs from "node:fs";
+import url from "node:url";
+import path from "node:path";
+
+const __dirname = url.fileURLToPath(new URL(".", import.meta.url)).slice(0, -1);
+const workspaceRoot = path.normalize(path.join(__dirname, "..", ".."));
+const rustTargetDir = path.join(workspaceRoot, "rust", "target");
+const rustWheelsDir = path.join(rustTargetDir, "wheels");
+
+const memoize = (f) => {
+    let val = undefined;
+    return () => {
+        if (typeof val !== "undefined") return val;
+        val = f();
+        return val;
+    };
+};
+
+export function getWorkspaceRoot() {
+    return workspaceRoot;
+}
+export function getRustTargetDir() {
+    return rustTargetDir;
+}
+export function getRustWheelsDir() {
+    return rustWheelsDir;
+}
+export function getEmscriptenWheelPath() {
+    const pspVersion = getWorkspacePackageJson().version;
+    const wheeljunk = "cp39-abi3-emscripten_3_1_58_wasm32";
```

Comment:
not sure if there's a good way to compute this (from the pyproject.toml? or Cargo.toml?)

Reply:
IMO we get this by listing any `.whl` target with "emscripten" in the triple name - and if there are two (or more), error with a message instructing the dev to run `pnpm run clean`.